### PR TITLE
Allow static imports of AssertJ and Mockito

### PR DESCRIPTION
**Pull request by pkoenig10**
## Before this PR
It's common to have a test utility project which is a test dependency in other projects. We currently allow static imports in tests (#240) but that does not apply to test utility projects since they are main sources.

This paper cut is aggravated by the `PreferAssertJ` check (#841) which static imports AssertJ methods. Doing this causes automated baseline upgrades to require manual intervention to successfully compile.

## After this PR
We allow static imports of AssertJ and Mockito methods.

### CODE REVIEW

**Comment by ferozco** [comment]
Given that the preferAsserJ migration is a onetime cost, I'm not sure if its worth widening the set of allowed static imports

**Comment by pkoenig10** [comment]
This isn't just for the AssertJ migration. It's cumbersome to not be able to use static imports in test utility classes just because they are part of the main source set in a test utility project.

These classes aren't going to be imported in non-test code so what's the downside of allowing them to be statically imported? These classes are almost universally statically imported in test code currently.

**Comment by carterkozak** [comment]
We allow static imports in test classes:
https://github.com/palantir/gradle-baseline/blob/develop/gradle-baseline-java-config/resources/checkstyle/checkstyle-suppressions.xml#L13

**Comment by schlosna** [comment]
@pkoenig10 can you use [Gradle 5.6+ test fixtures](https://docs.gradle.org/5.6/release-notes.html#test-fixtures-for-java-projects)?

**Comment by pkoenig10** [comment]
@carterkozak this is utility class used in tests, but it is not in a test source set. It is in a separate project that is imported as a test dependency in multiple other projects.

@schlosna test fixtures sound like exactly what I'm looking for, thanks for pointing that out!

I still think there is value in allowing these classes to be statically imported. Not everyone is going to have the ability or resources to migrate to test fixtures and the downside of allowing these classes to be imported is low.

**Review by iamdanfox — APPROVED** [review]
I just found an excavator assertj PR which was blocked by this checkstyle rule, because they'd defined a TestRule in their foo-ete project, in the src/main/java directory.

I think whitelisting the assertThat one is low regret

**Comment by pkoenig10** [comment]
@dfox I edited the changelog-app comment but it doesn't appear to be generating a changelog. Maybe you can do it?

**Comment by iamdanfox** [comment]
I think it's because this PR came from your personal fork, and changelog-app's repo-specific doesn't have perms on your fork sadly.  We've recently granted all palantirians write access to our OSS repos to workaround this.

**Comment by carterkozak** [comment]
changelog bot... bump?

**Comment by pkoenig10** [comment]
Moved to #915

**Pull request closed without merging**
